MSAL/src/instance/MSALAccountsProvider.m

```diff
@@ -238,6 +244,8 @@ - (MSALAccount *)accountForParameters:(MSALAccountEnumerationParameters *)parame
         [self addMSALAccount:externalAccount toSet:resultAccounts claims:accountClaims];
     }
     
+    resultAccounts = [self filterSignedOutAccounts:resultAccounts];
```

Comment:
this will also loop through external accounts which have been already filtered. Also, we should only filter out signed out accounts, if MSALAccountEnumerationParameters.returnOnlySignedInAccounts is true. Otherwise, skip this operation.

Reply:
make sense, addressed :)